src/connector/src/sink/file_sink/mod.rs

```diff
@@ -0,0 +1,230 @@
+    async fn barrier(&mut self, is_checkpoint: bool) -> Result<()> {
+        if is_checkpoint && let Some(sink_writer) = self.sink_writer.take() {
+            match sink_writer {
+                FileWriterEnum::ParquetFileWriter(w) => {
+                    let _ = w.close().await?;
+                }
+                FileWriterEnum::FileWriter(mut w) => w.close().await?,
+            };
+        }
+
+        Ok(())
+    }
+
+    async fn update_vnode_bitmap(&mut self, _vnode_bitmap: Arc<Bitmap>) -> Result<()> {
+        Ok(())
+    }
+}
+
+impl OpenDalSinkWriter {
+    pub fn new(
+        operator: Operator,
+        write_path: &str,
+        rw_schema: Schema,
+        pk_indices: Vec<usize>,
+        is_append_only: bool,
+        executor_id: u64,
+        encode_type: SinkEncode,
+    ) -> Result<Self> {
+        let arrow_schema = convert_rw_schema_to_arrow_schema(rw_schema)?;
+        Ok(Self {
+            schema: Arc::new(arrow_schema),
+            write_path: write_path.to_string(),
+            pk_indices,
+            operator,
+            sink_writer: None,
+            is_append_only,
+            epoch: None,
+            executor_id,
+            encode_type,
+        })
+    }
+
+    async fn create_object_writer(&mut self, epoch: u64) -> Result<OpendalWriter> {
+        // Todo: specify more file suffixes based on encode_type.
+        let suffix = match self.encode_type {
+            SinkEncode::Json => "json",
+            SinkEncode::Parquet => "parquet",
+            _ => unimplemented!(),
+        };
+
+        // Note: sink decoupling is not currently supported, which means that output files will not be batched across checkpoints.
+        // The current implementation writes files every time a checkpoint arrives, so the naming convention is `epoch + executor_id + .suffix`.
+        let object_name = format!(
+            "{}/epoch_{}_executor_{}.{}",
+            self.write_path, epoch, self.executor_id, suffix,
+        );
+        Ok(self
+            .operator
+            .writer_with(&object_name)
+            .concurrent(8)
+            .buffer(SINK_WRITE_BUFFER_SIZE)
+            .await?)
+    }
+
+    async fn create_sink_writer(&mut self, epoch: u64) -> Result<()> {
+        let object_writer = self.create_object_writer(epoch).await?;
+        match self.encode_type {
+            SinkEncode::Parquet => {
+                let props = WriterProperties::builder();
+                self.sink_writer = Some(FileWriterEnum::ParquetFileWriter(
+                    AsyncArrowWriter::try_new(
+                        object_writer,
+                        self.schema.clone(),
+                        SINK_WRITE_BUFFER_SIZE,
+                        Some(props.build()),
+                    )?,
+                ));
+            }
+            SinkEncode::Json => {
+                self.sink_writer = Some(FileWriterEnum::FileWriter(object_writer));
+                unimplemented!();
```

Comment:
ditto

Reply:
Only the type defined in https://github.com/risingwavelabs/risingwave/blob/ac20951e584ff149ea494f06a8dc68810f8e3547/src/frontend/src/handler/create_sink.rs#L777 can be parsed to the backend,otherwise, an error will be reported directly on the frontend.
Let's only introduce the parquet format in this PR, and leave json and other formats for later PRs.
Same below for this kind of unreachable arm.